pkgs/tools/misc/toybox/default.nix

```diff
@@ -0,0 +1,67 @@
+{
+  stdenv, lib, fetchFromGitHub, buildPackages,
+  enableStatic ? false,
+  enableMininal ? false,
+  extraConfig ? ""
+}:
+stdenv.mkDerivation rec {
+  name = "toybox-${version}";
+  version = "0.8.0";
+
+  src = fetchFromGitHub {
+    owner = "landley";
+    repo = "toybox";
+    rev = "b11e1206b4193dd46b4c40c333175fceadd409df";
+    sha256 = "0pghxf2ihx79rmqqgg61850m6kcnck51nn1ng5x20vd0wfh6gdl6";
+  };
+
+  depsBuildBuild = [ buildPackages.stdenv.cc ];
+  buildInputs = lib.optionals enableStatic [ stdenv.cc.libc stdenv.cc.libc.static ];
+
+  postPatch = "patchShebangs scripts";
+
+  configurePhase = ''
+    make ${if enableMininal then "allnoconfig" else if stdenv.isFreeBSD then "freebsd_defconfig" else if stdenv.isDarwin then "macos_defconfig" else "defconfig"}
+
+    function parseconfig {
+      while read LINE; do
+        NAME="$(echo "$LINE" | sed -e 's/^\([^ =]*\).*$/\1/')"
+        OPTION="$(echo "$LINE" | sed -e 's/[^ =]*[ =]\([^ =]*\)$/\1/')"
+
+        if ! [[ "$NAME" =~ ^CONFIG_ ]]; then continue; fi
+
+        echo "parseconfig: removing $NAME"
+        sed -i /$NAME'\(=\| \)'/d .config
+
+        if [ "$OPTION" = 'n' ]; then
+          echo "parseconfig: unsetting $NAME"
+          echo "# $NAME is not set" >> .config
+        else
+          echo "parseconfig: setting $NAME=$OPTION"
+          echo "$NAME=$OPTION" >> .config
+        fi
+      done
+    }
+
+    cat << EOF | parseconfig
+    ${extraConfig}
```

Comment:
Use
```nix
{
  # ...
  inherit extraConfig;
  passAsFile = [ "extraConfig" ];
}
```

instead. And then you can use this instead which works fairly well as far as I can see, because the build uses whatever value is set first:
```bash
make ${if enable...

cat $extraConfigPath .config > .config-
mv .config- .config

make oldconfig
```

Reply:
Integrated this; thanks for the code! :-)